pkg/gatherers/clusterconfig/machines.go

```diff
@@ -0,0 +1,66 @@
+package clusterconfig
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/openshift/insights-operator/pkg/utils/anonymize"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog/v2"
+)
+
+func (g *Gatherer) GatherMachine(ctx context.Context) ([]record.Record, []error) {
+	dynamicClient, err := dynamic.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return gatherMachine(ctx, dynamicClient)
+}
+
+func gatherMachine(ctx context.Context, dynamicClient dynamic.Interface) ([]record.Record, []error) {
+	gvr := schema.GroupVersionResource{Group: "machine.openshift.io", Version: "v1beta1", Resource: "machines"}
+	machines, err := dynamicClient.Resource(gvr).List(ctx, metav1.ListOptions{})
+	if errors.IsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, []error{err}
+	}
+	var records []record.Record
```

Comment:
Lint is complaining about this block of code, 'cause it duplicates [this](https://github.com/openshift/insights-operator/blob/1f10ef9673a311c232d51aa68dbac31cfb56e65b/pkg/gatherers/clusterconfig/machine_sets.go#L53). Maybe we just need to move it as a function so it can be used in both places.